Samples/Tutorial Samples/YOLOv4ObjectDetection/YOLOv4ObjectDetection/MainPage.xaml.cs

```diff
@@ -165,141 +88,63 @@ private async Task InitCameraAsync()
+                    watch.Start();
                 }
-                if (j==final_detections.Count)
+
+                processing = true;
+                var frame = new VideoFrame(Windows.Graphics.Imaging.BitmapPixelFormat.Bgra8, (int)WebCam.Width, (int)WebCam.Height);
+                await _media_capture.GetPreviewFrameAsync(frame);
+                var results = await _model.EvaluateFrame(frame);
+                await DrawBoxes(results, frame);
+                count++;
+                if (watch.ElapsedMilliseconds > 1000)
                 {
-                    final_detections.Add(detections[i]);
+                    ShowStatus(string.Format("{0} fps", count));
+                    count = 0;
+                    watch.Restart();
                 }
+            } 
+            finally
+            {
+                processing = false;
             }
-            return final_detections;
         }
 
-        // parse the result from WinML evaluation results to self defined object struct
-        private List<DetectionResult> ParseResult(float[] results)
+        private float Sigmoid(float val)
         {
-            int c_values = 84;
-            int c_boxes = results.Length / c_values;
-            float confidence_threshold = 0.5f;
-            List<DetectionResult> detections = new List<DetectionResult>();
-            this.OverlayCanvas.Children.Clear();
-            for (int i_box = 0; i_box < c_boxes; i_box++)
-            {
-                float max_prob = 0.0f;
-                int label_index = -1;
-                for (int j_confidence = 4; j_confidence < c_values; j_confidence++)
-                {
-                    int index = i_box * c_values + j_confidence;
-                    if (results[index] > max_prob)
-                    {
-                        max_prob = results[index];
-                        label_index = j_confidence - 4;
-                    }
-                }
-                if (max_prob > confidence_threshold)
-                {
-                    List<float> bbox = new List<float>();
-                    bbox.Add(results[i_box * c_values + 0]);
-                    bbox.Add(results[i_box * c_values + 1]);
-                    bbox.Add(results[i_box * c_values + 2]);
-                    bbox.Add(results[i_box * c_values + 3]);
-
-                    detections.Add(new DetectionResult()
-                    {
-                        label = _labels[label_index],
-                        bbox = bbox,
-                        prob = max_prob
-                    });
-                }
-            }
-            return detections;
+            var x = (float)Math.Exp(val);
+            return x / (1.0f + x);
         }
 
         // draw bounding boxes on the output frame based on evaluation result
-        private async Task DrawBoxes(float[] results, VideoFrame frame)
+        private async Task DrawBoxes(List<Model.DetectionResult> detetions, VideoFrame frame)
```

Comment:
typo - can you correct this to "detections" or "detectionResults"?

Reply:
woops, fixed.